src/system.rs

```diff
@@ -447,4 +635,171 @@ mod schema_system_tests {
         let schema_system_authorities = schema_system.authorities();
         assert_eq!(2, schema_system_authorities.len());
     }
+
+    #[test]
+    fn schema_system_map_authority_with_type_alias_import_test() {
+        // map with (id, ion content)
+        let map_authority: HashMap<String, String> = [
+            (
+                "sample_number.isl",
+                r#"
+                    schema_header::{
+                      imports: [{ id: "sample_decimal.isl", type: my_decimal, as: other_decimal }],
+                    }
+                    
+                    type::{
+                      name: my_int,
+                      type: int,
+                    }
+                    
+                    type::{
+                      name: my_number,
+                      all_of: [
+                        my_int,
+                        other_decimal,
+                      ],
+                    }
+                    
+                    schema_footer::{
+                    }
+                "#,
+            ),
+            (
+                "sample_decimal.isl",
+                r#"
+                    schema_header::{
+                      imports: [],
+                    }
+                    
+                    type::{
+                      name: my_decimal,
+                      type: decimal,
+                    }
+                    
+                    type::{
+                      name: my_string,
+                      type: string,
+                    }
+                    
+                    schema_footer::{
+                    }
+                "#,
+            ),
+        ]
+        .iter()
+        .map(|(id, ion)| (id.to_owned().to_owned(), ion.to_owned().to_owned()))
+        .collect();
+        let mut schema_system =
+            SchemaSystem::new(vec![Box::new(MapDocumentAuthority::new(map_authority))]);
```

Comment:
Food for thought: maybe `MapDocumentAuthority`'s constructor should take an `IntoIterator<Item=(&str, &str)>`? Users wouldn't have to call `map`/`collect`, and the implementation detail of which kind of map it's using would stay hidden.